internal/gensample/plugin.go

```diff
@@ -0,0 +1,78 @@
+// Copyright 2019 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package gensample
+
+import (
+	"path"
+	"path/filepath"
+	"strings"
+
+	plugin "github.com/golang/protobuf/protoc-gen-go/plugin"
+	"github.com/googleapis/gapic-generator-go/internal/errors"
+)
+
+const (
+	paramError = "need parameter in format: go-gapic-package=client/import/path;packageName"
+)
+
+// PluginEntry is the entry point of SampleGen as a protoc plugin, together with CodeGen
+func PluginEntry(genReq *plugin.CodeGeneratorRequest) (*plugin.CodeGeneratorResponse, error) {
```

Comment:
We are having one protoc plugin only that can generate one or both of the library and the sample, right? I thought you were keeping samplegen as its own binary, but we're having only one plugin?

Reply:
Yeah, samplegen can continue to run as a standalone go binary. (We already have this, and installing/running go binaries is easy, so why not?)

But when running the microgenerator as a plugin, if sample configs are specified, samples are also generated. (The setting is very similar to Python micro).
 
In an ideal world, when all APIs are annotated, DPEs just need to run the microgenerator (as a docker or a plugin), and can get all the code and samples for an API.

Reply:
(That reminds me that I should have @software-dov for review, to ensure the consistency between Go and Python micro generators)